src/runtime/virtcontainers/container.go

```diff
@@ -529,6 +529,12 @@ func (c *Container) mountSharedDirMounts(ctx context.Context, sharedDirMounts, i
 		sharedDirMounts[sharedDirMount.Destination] = sharedDirMount
 	}
 
+	// Start the event loop to watch for file change notifications
+	// The event loop will only start if there are watches added
+	if c.sandbox.fsShare != nil {
+		go c.sandbox.fsShare.StartFileEventWatcher(ctx)
+	}
```

Comment:
Should we put this in the same deferred call created at the beginning of the function?

Because if we have early returns (e.g., line 477 above), then we end up not starting the watcher.

Reply:
I've moved this up into the defer as suggested.